platform-sdk/swirlds-base/src/main/java/com/swirlds/base/utility/Network.java

```diff
@@ -0,0 +1,100 @@
+ */
+public final class Network {
+
+    /**
+     * Private constructor to prevent utility class instantiation.
+     */
+    private Network() {}
+
+    /**
+     * Evaluates a given {@code hostname} to ensure it is resolvable to one or more valid IPv4 or IPv6 addresses.
+     * Supports domain name fragments, fully qualified domain names (FQDN), and IP addresses.
+     *
+     * <p>
+     * This method includes retry logic to handle slow DNS queries due to network conditions, slow DNS record
+     * updates/propagation, and/or intermittent network connections.
+     *
+     * <p>
+     * This overloaded method uses a default of 20 attempts with a 2-second delay between each attempt.
+     *
+     * @param name the domain name fragment, fully qualified domain name (FQDN), or IP address to be resolved.
+     * @return true if the name resolves to one or more valid IP addresses; otherwise false.
+     * @see #isNameResolvable(String, int, int)
+     */
+    public static boolean isNameResolvable(@NonNull final String name) {
+        return isNameResolvable(name, 20, 2_000);
+    }
+
+    /**
+     * Evaluates a given {@code hostname} to ensure it is resolvable to one or more valid IPv4 or IPv6 addresses.
+     * Supports domain name fragments, fully qualified domain names (FQDN), and IP addresses.
+     *
+     * <p>
+     * This method includes retry logic to handle slow DNS queries due to network conditions, slow DNS record
+     * updates/propagation, and/or intermittent network connections.
+     *
+     * @param name        the domain name fragment, fully qualified domain name (FQDN), or IP address to be resolved.
+     * @param maxAttempts the maximum number of retry attempts.
+     * @param delayMs     the delay between retry attempts.
+     * @return true if the name resolves to one or more valid IP addresses; otherwise false.
+     */
+    public static boolean isNameResolvable(@NonNull final String name, final int maxAttempts, final int delayMs) {
+        Objects.requireNonNull(name, "name must not be null");
+        try {
+            return Retry.check(Network::isNameResolvableInternal, name, maxAttempts, delayMs);
+        } catch (InterruptedException ignored) {
+            Thread.currentThread().interrupt();
+            return false;
+        } catch (ExecutionException ignored) {
+            return false;
+        }
+    }
+
+    /**
+     * Internal implementation used by the {@link #isNameResolvable(String, int, int)} method.
+     *
+     * @param name the domain name fragment, fully qualified domain name (FQDN), or IP address to be resolved.
+     * @return true if the name resolves to one or more valid IP addresses; otherwise false.
+     */
+    private static boolean isNameResolvableInternal(@NonNull final String name) {
+        Objects.requireNonNull(name, "name must not be null");
+        try {
+            final InetAddress[] addresses = InetAddress.getAllByName(name);
```

Comment:
Just wondering if `InetAddress.getByName` would suffice here instead of  `InetAddress.getAllByName(name)`

Also I can see `InetAddress.getByName()` is used in the code (in config.txt parser) to resolve to an IP, so it would make the code consistent.